src/openms/source/ANALYSIS/ID/IonIdentityMolecularNetworking.cpp

```diff
@@ -214,7 +215,7 @@ namespace OpenMS
     if (!consensus_map[0].metaValueExists(Constants::UserParam::IIMN_ROW_ID)) return;
 
     // generate unordered map with feature id and partner feature ids
-    std::unordered_map<size_t, std::set<size_t>> feature_partners; // map<feature_index, partner_feature_indices>
+    std::map<size_t, std::set<size_t>> feature_partners; // map<feature_index, partner_feature_indices>
```

Comment:
this could change the algorithm speed a lot as it changes access from constant time to log time.
If it is not an issue I think it is fine.

Reply:
How should we deal with the tests? They are not reproducible with the unordered map.

Reply:
if memory is not a big deal you could just do a
map<...> sorted(feature_partners.begin(), feature_partners.end());
and then use this map instead of for (const auto& entry: feature_partners)
before writing out results.
If memory is an issue you could probably swap out the values so these don't get copied:
for (auto& [key, value] : feature_partners)
{
  sorted[key].swap(value);
}

Reply:
I remember Julianus(?) adding some code to make the ordering consistent across platforms...
Cant find it ad hoc...

Reply:
This was just for Random numbers I think.

Reply:
True.

does the result need to be sorted for the user, or merely the test?
If its just the test, then why not capture the output (as string?) and sort that instead of adding a conversion penalty in the production code. ??
Or get fancy and use a templated function, where you inject the type of container as template parameter.

Reply:
For the user it doesn't matter if the result is sorted, so adjusting the tests might be an option.

Reply:
Just trying the solution suggested by Timo as well to see if that works.

Reply:
Thanks for the suggestions. Have created a sorted map while swapping the values to save memory. This is a good solution since even if the sorting is not strongly required for the user, it gives a better readable output.